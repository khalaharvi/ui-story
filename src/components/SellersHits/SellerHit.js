import React from 'react';
import PropTypes from 'prop-types';

import { capitalizeFirstLetter } from 'utils';
import Image from 'components/Image';
import { cx } from './SellersHits';

const getSellerTypeDisplayName = types =>
  types
    .map(type =>
      type.split('-').map(typeChunk =>
        capitalizeFirstLetter(typeChunk)).join(' '))
    .join(', ');

// TO DO: add fallback images
const SellerHit = ({ hit, transformHitImageUrl }) => (
  <div to={hit.product_path} className={cx('list-item-link')}>
    <div className={cx('list-item-img')}>
      <Image
        src={
          transformHitImageUrl ?
          transformHitImageUrl(hit.primary_image_url) :
          hit.primary_image_url
        }
        alt={hit.name}
      />
    </div>
    <b className={cx('list-item-name')}>{hit.name}</b>
    <i className={cx('list-item-types')}>{getSellerTypeDisplayName(hit.seller_types)}</i>
  </div>
);

SellerHit.propTypes = {
  transformHitImageUrl: PropTypes.func,
  hit: PropTypes.shape({
    slug: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    seller_types: PropTypes.arrayOf(PropTypes.string).isRequired,
    primary_image_url: PropTypes.string.isRequired,
    primary_type: PropTypes.string.isRequired
  })
};

export default SellerHit;
