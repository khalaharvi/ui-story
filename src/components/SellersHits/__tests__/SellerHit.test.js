import React from 'react';
import { shallow } from 'enzyme';

import SellerHit from '../SellerHit';

describe('<SellerHit />', () => {
  it('renders correctly', () => {
    const hit = {
      name: 'Brand Name',
      primary_image_url: 'https://placehold.it/370x220',
      slug: '/brand-name',
      seller_types: ['interior-designer'],
      primary_type: 'interior-designer'
    };

    const wrapper = shallow(<SellerHit hit={hit} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with transformHitImageUrl', () => {
    const hit = {
      name: 'Brand Name',
      primary_image_url: 'https://placehold.it/370x220',
      slug: '/brand-name',
      seller_types: ['interior-designer'],
      primary_type: 'interior-designer'
    };

    const wrapper = shallow(
      <SellerHit
        hit={hit}
        transformHitImageUrl={url => url}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
