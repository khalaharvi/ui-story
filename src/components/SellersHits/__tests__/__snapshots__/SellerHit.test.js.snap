// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SellerHit /> renders correctly 1`] = `
<div
  className="dh-sellers-hits-list-item-link"
>
  <div
    className="dh-sellers-hits-list-item-img"
  >
    <Image
      alt="Brand Name"
      src="https://placehold.it/370x220"
    />
  </div>
  <b
    className="dh-sellers-hits-list-item-name"
  >
    Brand Name
  </b>
  <i
    className="dh-sellers-hits-list-item-types"
  >
    Interior Designer
  </i>
</div>
`;

exports[`<SellerHit /> renders correctly with transformHitImageUrl 1`] = `
<div
  className="dh-sellers-hits-list-item-link"
>
  <div
    className="dh-sellers-hits-list-item-img"
  >
    <Image
      alt="Brand Name"
      src="https://placehold.it/370x220"
    />
  </div>
  <b
    className="dh-sellers-hits-list-item-name"
  >
    Brand Name
  </b>
  <i
    className="dh-sellers-hits-list-item-types"
  >
    Interior Designer
  </i>
</div>
`;
